code/game/machinery/Sleeper.dm

```diff
@@ -272,6 +271,14 @@
 /obj/machinery/sleeper/MouseDrop_T(var/mob/target, var/mob/user)
 	if(user.stat || user.lying || !Adjacent(user) || !target.Adjacent(user)|| !ishuman(target))
 		return
+
+	var/mob/living/L = target
+	var/bucklestatus = L.bucklecheck(user)
+	if(!bucklestatus)
+		return
+	if(bucklestatus == 2)
+		var/obj/structure/LB = L.buckled_to
+		LB.user_unbuckle(user)
```

Comment:
I still don't understand the need for a thing to return 2 if the user is a robot. Why does `bucklecheck` need to return anything other than a bool?

Can you explain what this snippet is doing that is important?